### PR TITLE
HAAR-345 hide email domain prefix

### PR DESCRIPTION
**Pull request by AlTurner-MOJ**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/EmailDomainService.kt [hunk 1]
```diff
@@ -15,24 +16,40 @@ class EmailDomainService(
   private val emailDomainRepository: EmailDomainRepository,
   private val emailDomainExclusions: EmailDomainExclusions,
 ) {
+  private val percent = "%"
```
Comment (brightonsbox) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/EmailDomainService.kt:19]: I think it's best practice with Kotlin to put such constants in a private companion object (e.g. https://github.com/ministryofjustice/hmpps-auth/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/security/UserStateAuthenticationFailureHandler.kt#L32 )

And to capitalise the name

**Review by brightonsbox — APPROVED** [review]
LGTM - just one minor comment about constants

File: src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/EmailDomainService.kt (hunk 1 above)
Comment (AlTurner-MOJ) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/oauth2server/service/EmailDomainService.kt]: Thanks Jonathan - I've changed that

**Review by Mjwillis — APPROVED** [review]
LGTM

**Review by brightonsbox — APPROVED** [review]
LGTM

**Pull request merged**